src/App.tsx

```diff
@@ -15,48 +16,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Name = 'Sort alphabetically',
+  Length = 'Sort by length',
+  Null = '',
+  Reverse = 'Reverse',
+}
+
+interface Buttons {
+  sortType: SortType,
+  buttonStyle: string,
+}
+
+const buttons: Buttons[] = [
+  { sortType: SortType.Name, buttonStyle: 'is-info' },
+  { sortType: SortType.Length, buttonStyle: 'is-success' },
+  { sortType: SortType.Reverse, buttonStyle: 'is-warning' },
+];
+
+const sortByParams = (
+  toSortArray: string[],
+  sortType: SortType,
+  isReversed: boolean,
+): string[] => {
+  let array: string[] = [...toSortArray];
+
+  array = !SortType.Null && array.sort((a: string, b: string) => {
+    switch (sortType) {
+      case SortType.Name:
+        return a.localeCompare(b);
+
+      case SortType.Length:
+        return a.length - b.length;
+
+      case SortType.Null:
+      default:
+        return 0;
+    }
+  });
+
+  return isReversed ? array.reverse() : array;
+};
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState<SortType>(SortType.Null);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  function reset(): void {
+    setSortType(SortType.Null);
+    setIsReversed(false);
+  }
+
```

Comment:
I'm not sure? but as i remember, we can not write type void if the function return nothing

Reply:
we can, but i want to